applications/GeoMechanicsApplication/custom_constitutive/coulomb_yield_surface.cpp

```diff
@@ -40,35 +45,41 @@ std::string GetCoulombHardeningTypeFrom(const Properties& rMaterialProperties)
 CoulombYieldSurface::KappaDependentFunction MakeFrictionAngleCalculator(const Properties& rMaterialProperties)
 {
     const auto hardening_type = GetCoulombHardeningTypeFrom(rMaterialProperties);
-
     if (hardening_type == "none") {
         return MakeConstantFunction(ConstitutiveLawUtilities::GetFrictionAngleInRadians(rMaterialProperties));
     }
-
+    if (hardening_type == "linear") {
+        return MakeLinearFunction(ConstitutiveLawUtilities::GetFrictionAngleInRadians(rMaterialProperties),
+                                  rMaterialProperties[GEO_FRICTION_ANGLE_FUNCTION_COEFFICIENTS](0));
```

Comment:
For accessing the value of a vector, we prefer to use `[]` rather than `()`, since it resembles the style used by the STL. When accessing an element requires multiple indices, using `()` is the way to go.
```suggestion
                                  rMaterialProperties[GEO_FRICTION_ANGLE_FUNCTION_COEFFICIENTS][0]);
```
I have the same suggestion for the other two factory functions.